lms/djangoapps/instructor/views/api.py

```diff
@@ -2707,10 +2707,129 @@ def start_certificate_regeneration(request, course_id):
+    """
+    Add/Remove students to/from certificate white list.
+
+    :param request: HttpRequest object
+    :param course_id: course identifier of the course for whom to add/remove certificates exception.
+    :param certificate_exception_id: id of CertificateWhitelist item, this will be provided in case of
+           deleting certificate exception, it will be empty if new certificate exception is being created.
+    :return: JsonResponse object with success/error message or certificate exception to sync on the frontend.
+    """
+    course_key = CourseKey.from_string(course_id)
+    try:
+        certificate_exception = json.loads(request.body or '{}')
+    except ValueError:
+        return JsonResponse({
+            'success': False,
+            'message': _('Invalid Json data')
+        }, status=400)
+
+    if request.method == 'POST':
+        try:
+            exception = add_certificate_exception(certificate_exception, course_key)
+        except ValueError as error:
+            return JsonResponse({'success': False, 'message': error.message}, status=400)
+        return JsonResponse(exception)
+
+    elif request.method == 'DELETE':
+        try:
+            remove_certificate_exception(course_key, certificate_exception_id)
+        except ValueError as error:
+            return JsonResponse({'success': False, 'message': error.message}, status=400)
+
+        return JsonResponse({})
+
+
+def add_certificate_exception(certificate_exception, course_key):
+    """
+    Add a certificate exception to CertificateWhitelist table. Raises ValueError in case of invalid data or if
+    the user specified in certificate_exception does not exist in the database.
+
+    :param certificate_exception: JSON representation of backbone CertificateException model
+    :param course_key: course key of the course for whom to add certificates exception
+    :return: CertificateWhitelist item in dict format containing all the keys required by
+             backbone model CertificateException.
+    """
+    user = certificate_exception.get('user_name', '') or certificate_exception.get('user_email', '')
+    if not user:
+        raise ValueError(_('Student username/email is required.'))
+    try:
+        db_user = get_user_by_username_or_email(user)
+    except ObjectDoesNotExist:
+        raise ValueError(_('Student (username/email={user}) does not exist').format(user=user))
+    except MultipleObjectsReturned:
+        raise ValueError(_('Multiple Students found with username/email={user}').format(user=user))
+
+    if CertificateWhitelist.objects.filter(user=db_user, course_id=course_key, whitelist=True).count() > 0:
+        raise ValueError(
+            _("Student (username/email={user_id} already in certificate exception list)").format(user_id=user)
+        )
+
+    certificate_white_list, created = CertificateWhitelist.objects.get_or_create(  # pylint: disable=unused-variable
```

Comment:
nit: If we use `__` instead of `created` here we can get rid of pylint's `unused-variable` violation.